### PR TITLE
Remove ramda's memoize function and use fast-memoize (#90)

### LINKED ISSUE #90: Remove ramda's memoize function and use fast-memoize instead
The numbers shown in https://github.com/caiogondim/fast-memoize.js for fast-memoize are far more performant than ramda's memoize function.

### PR DESCRIPTION
**Pull request by aimenbatool**
Replaced memoizeWith function with memoize in all places. 

Resolves  #90.

### CODE REVIEW

**Comment by aimenbatool** [comment]
Hi @helfi92, According to my understanding `memoize` function takes 3 parameters and the third parameter is a serializer. Which is mentioned in the doc you gave and in one of the file may be implemented by you. 

Here is what I figured out to do

- `import memoize from 'fast-memoize'`

- Add 3rd serialize paramter in memoize function 

```
serializer: ([healthData, sortBy, sortDirection]) => {
        const ids = sorted(healthData.running);

        return `${ids.join('-')}-${sortBy}-${sortDirection}`;
      },
```
Changing the healthData variable according to each function in all files. 

Does this make sense to you? Please, guide. 
Thanks

**Comment by aimenbatool** [comment]
> The change will look like:
> 
> #### Before
> ```js-jsx
> import { memoizeWith } from 'ramda';
> 
> memoizeWith(serializerFunction, actualFunction)
> ```
> 
> #### After
> ```js-jsx
> import memoize from 'fast-memoize';
> 
> memoize(actualFunction, {
>   serializer: serializerFunction
> });
> ```

Perfect. I got it. Let me apply the change to all places.

**Comment by helfi92** [comment]
Thank you. I will review this soon.

File: src/components/CachePurgesTable/index.jsx [hunk 1]
```diff
@@ -69,6 +65,13 @@ export default class CachePurgesTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (cachePurgesConnection, sortBy, sortDirection) => {
```
Comment (helfi92) [inline src/components/CachePurgesTable/index.jsx]: This needs to be in an array: `[cachePurgesConnection, sortBy, sortDirection]`.

File: src/components/ClientsTable/index.jsx [hunk 2]
```diff
@@ -62,6 +58,13 @@ export default class ClientsTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (clientsConnection, sortBy, sortDirection) => {
```
Comment (helfi92) [inline src/components/ClientsTable/index.jsx]: This needs to be in an array.

File: src/components/IndexTaskNamespaceTable/index.jsx [hunk 3]
```diff
@@ -85,6 +81,13 @@ export default class IndexTaskNamespaceTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (connection, sortBy, sortDirection) => {
```
Comment (helfi92) [inline src/components/IndexTaskNamespaceTable/index.jsx]: This needs to be in an array.

File: src/components/RoleScopesTable/index.jsx [hunk 4]
```diff
@@ -87,6 +83,12 @@ export default class RoleScopesTable extends Component {
       );
 
       return selectedScope ? extractRoles(roles) : extractExpandedScopes(roles);
+    },
+    {
+      serializer: (roles, searchMode, selectedScope, searchProperty) =>
```
Comment (helfi92) [inline src/components/RoleScopesTable/index.jsx]: This needs to be in an array.

File: src/components/RolesTable/index.jsx [hunk 5]
```diff
@@ -80,6 +68,13 @@ export default class RolesTable extends Component {
           return sort(firstElement, secondElement);
         })
       )(filteredRoles);
+    },
+    {
+      serializer: (roles, sortBy, sortDirection, searchTerm) => {
```
Comment (helfi92) [inline src/components/RolesTable/index.jsx]: This needs to be in an array.

File: src/components/SecretsTable/index.jsx [hunk 6]
```diff
@@ -84,6 +80,13 @@ export default class SecretsTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (secretsConnection, sortBy, sortDirection) => {
```
Comment (helfi92) [inline src/components/SecretsTable/index.jsx]: This needs to be in an array.

File: src/components/WorkerTypesTable/index.jsx [hunk 7]
```diff
@@ -154,6 +147,18 @@ export default class WorkerTypesTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (
+        workerTypesConnection,
+        awsProvisionerWorkerTypeSummaries,
+        sortBy,
+        sortDirection
```
Comment (helfi92) [inline src/components/WorkerTypesTable/index.jsx]: This needs to be in an array.

File: src/components/WorkersTable/index.jsx [hunk 8]
```diff
@@ -83,6 +79,13 @@ export default class WorkersTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: (workersConnections, sortBy, sortDirection) => {
```
Comment (helfi92) [inline src/components/WorkersTable/index.jsx]: This needs to be in an array.

**Comment by aimenbatool** [comment]
Oh! I missed it. Applied in a few but forgot in most. Let me make changes.

**Comment by aimenbatool** [comment]
Applied the required changes. Please check the latest commit.

**Comment by helfi92** [comment]
Looks good. I will pull down the code to make some tests.

File: src/components/WorkerTypesTable/index.jsx [hunk 9]
```diff
@@ -154,6 +147,13 @@ export default class WorkerTypesTable extends Component {
           return sort(firstElement, secondElement);
         }),
       };
+    },
+    {
+      serializer: ([workerTypesConnection, sortBy, sortDirection]) => {
```
Comment (helfi92) [inline src/components/WorkerTypesTable/index.jsx]: `awsProvisionerWorkerTypeSummaries` is missing after `workerTypesConnection`.

Comment (aimenbatool) [reply src/components/WorkerTypesTable/index.jsx]: > `awsProvisionerWorkerTypeSummaries` is missing after `workerTypesConnection`.

It was giving me error that `awsProvisionerWorkerTypeSummaries` is defined but not used. It was not compiling code successfully so I removed it. Do you want me to add ignoring the error?

Comment (aimenbatool) [reply src/components/WorkerTypesTable/index.jsx]: screenshot from 2018-09-27 19-46-09


Check this error. I can add it back If you want.

Comment (helfi92) [reply src/components/WorkerTypesTable/index.jsx]: You can supress this warning by doing:

```jsx
serializer: ([
  workerTypesConnection,
  // eslint-disable-next-line no-unused-vars
  awsProvisionerWorkerTypeSummaries,
  sortBy,
  sortDirection,
])
```

**Comment by aimenbatool** [comment]
Updated. :) I hope it will work now.

**Pull request merged**